src/main/kotlin/com/openlattice/entitysets/EntitySetsApi.kt

```diff
@@ -204,8 +206,20 @@ interface EntitySetsApi {
      *
      * @return Set of entity key ids that will expire before the specified date time
      */
-    @POST(BASE + ALL + ID_PATH + EXPIRATION_PATH )
-    fun getExpiringEntitiesFromEntitySet(@Path(ID) entitySetId: UUID,
-                                         @Body dateTime: String): Set<UUID>
+    @POST(BASE + ALL + ID_PATH + EXPIRATION_PATH)
+    fun getExpiringEntitiesFromEntitySet(
+            @Path(ID) entitySetId: UUID,
+            @Body dateTime: String
+    ): Set<UUID>
 
+    /**
+     * Used to set the new partitions for an entity set. This will shuffle corresponding ids, edges, and data table rows
+     * for the entity set.
+     *
+     * @param entitySetId The id of the entity set to update.
+     * @param partitions The partitions to set.
+     * @return The job id for the repartioning job.
+     */
+    @PUT(BASE + ID_PATH + PARTITIONS_PATH)
+    fun setPartitions(@Path(ID) entitySetId: UUID, @Body partitions: Set<Int>): UUID
```

Comment:
I think this should definitely go in the Admin API

Reply:
That's where it was originally, but it felt out of place.

Reply:
Isn't this the call that triggers the repartition job? I definitely think that should be part of an admin only api or at least require admin creds. 
So yeah its fine here but I think the doc should show that it requires admin creds

Reply:
It does require admin only on the datastore side and I will update the doc. There are a number of other admin only APIs located (addConnection on OrganizationsController, a bunch of calls in IndexingController, CollectionsController, etc). I do think we need to rationalize what we're putting in there, because it's all over the place (SMS for orgs, random counting operations, cache operations, etc). My first stab at it would be to limit it to stuff that is operational (resetting caches, read/write of system wide configs (future example is registering new datastores), managing notification templates, etc. 

tl;dr We've been haphazard about where we place admin only API calls, but placing them in a natural locations, seems better than the random stuff that's been dumped on AdminController, but we should have a better heuristic.

Reply:
https://jira.openlattice.com/browse/LATTICE-2535